calliope/backend/pyomo/model.py

```diff
@@ -101,32 +102,23 @@ def generate_model(model_data):
 
     # Constraints
     constraints_to_add = [
-        'energy_balance.load_constraints',
-        'dispatch.load_constraints',
-        'network.load_constraints',
-        'costs.load_constraints',
-        'policy.load_constraints',
-        'milp.load_constraints',
-        'group.load_constraints'
+        i.split('.py')[0] for i in os.listdir(constraints.__path__[0])
+        if '__' not in i
     ]
 
-    if backend_model.__calliope_run_config['mode'] != 'operate':
-        constraints_to_add.append('capacity.load_constraints')
-
-    if hasattr(backend_model, 'loc_techs_conversion'):
-        constraints_to_add.append('conversion.load_constraints')
-
-    if hasattr(backend_model, 'loc_techs_conversion_plus'):
-        constraints_to_add.append('conversion_plus.load_constraints')
+    # The list is sorted to ensure that some constraints are added after pyomo
+    # expressions have been created in other constraint files.
+    # Ordering is given by the number assigned to the variable ORDER within each
+    # file (higher number = added later).
+    constraints_to_add.sort(
+        key=lambda x: load_function('calliope.backend.pyomo.constraints.' + x + '.ORDER')
```

Comment:
This would blow up if a constraint file does not have ORDER, right? Maybe should files without an ORDER have a very large ORDER number by default and a warning be raised about that?

Reply:
And I wonder whether this warrants a test?

Reply:
RE files without order. I can add in the warning and the forced order position.
RE tests, I was trying to think of how to do a test on this, but couldn't create a viable one. Ideally I would capture the debug comment about which files have been loaded, but I couldn't get that to pop up when using the pytest [logging capture function](https://docs.pytest.org/en/latest/logging.html).

Reply:
Done